src/ncp/ncp_base_radio.cpp

```diff
@@ -481,6 +482,11 @@ otError NcpBase::DecodeStreamRawTxRequest(otRadioFrame &aFrame)
     SuccessOrExit(mDecoder.ReadUint32(aFrame.mInfo.mTxInfo.mTxDelay));
     SuccessOrExit(mDecoder.ReadUint32(aFrame.mInfo.mTxInfo.mTxDelayBaseTime));
     SuccessOrExit(mDecoder.ReadUint8(aFrame.mInfo.mTxInfo.mRxChannelAfterTxDone));
+    if ((error = mDecoder.ReadInt8(aFrame.mInfo.mTxInfo.mTxPower)) == OT_ERROR_PARSE)
+    {
+        // Intentionally ignore the error for backward compatibility.
+        error = OT_ERROR_NONE;
+    }
```

Comment:
We set `mTxPower` above and then here indirectly assume that `ReadInt8()` will not update it when it returns an error (which should generally be the case, but is not something we explicitly mention and guarantee).

I would suggest either using `mDecoder.IsAllRead()` (or `mDecoder.IsAllReadInStruct()`) before calling `ReadInt8()` or alternate way can be to set `mTxPower` after `ReadUint8()` is called and when it returns an `error`.

Reply:
Good point. How about updating the API document `Read*()` to promise it won't change the value if there's an error? This way, the code would look cleaner. Otherwise, we have to add `IsAllRead()`  before reading all new fields. Thoughts?

Reply:
Sounds good to me.

Actually all the other fields in the code before this one are also treated as optional, so we can follow the same pattern (`SuccessOrExit()` without setting `error`).
```
    SuccessOrExit(mDecoder.ReadUint32(aFrame.mInfo.mTxInfo.mTxDelay));
    SuccessOrExit(mDecoder.ReadUint32(aFrame.mInfo.mTxInfo.mTxDelayBaseTime));
    SuccessOrExit(mDecoder.ReadUint8(aFrame.mInfo.mTxInfo.mRxChannelAfterTxDone));
```